src/main/java/dog/pawbook/model/managedentity/owner/UniqueEntityList.java

```diff
@@ -189,6 +173,13 @@ public void setEntities(List<Pair<Integer, Entity>> entities) {
         return internalUnmodifiableList;
     }
 
+    /**
+     * Returns the backing list as an {@code ObservableList}.
+     */
+    public ObservableList<Pair<Integer, Entity>> asObservableList() {
+        return internalList;
+    }
+
```

Comment:
Okay I understand this portion now, it might seem fine to do this for now, but the issue is that this codebase isn't very small, having equivalent api will confuse others that are not so familiar with this portion, and any direct modification is now possible since Java passes by reference, it will be difficult to track down bugs if someone modifies it and violates the assumptions we have about unique entities etc.

Reply:
Removed